helix-term/src/commands.rs

```diff
@@ -4647,8 +4647,13 @@ fn shell_prompt(cx: &mut Context, prompt: Cow<'static, str>, behavior: ShellBeha
 }
 
 fn suspend(_cx: &mut Context) {
-    #[cfg(not(windows))]
-    signal_hook::low_level::raise(signal_hook::consts::signal::SIGTSTP).unwrap();
+    if *crate::application::ENABLE_SIGTSTP
+        .get()
+        .expect("ENABLE_SIGTSTP flag must be set")
+    {
+        #[cfg(not(windows))]
+        signal_hook::low_level::raise(signal_hook::consts::signal::SIGTSTP).unwrap();
+    }
```

Comment:
A few things here:

Since the inside of the if statement is not run on windows, no need to run this whole expression on windows. There isn't so much benefit in this case of using `#[cfg(not(windows))]` over `cfg!(not(windows))`, it's better to be idiomatic and have it in the if condition IMO.

The `expect` will panic on `C-z` if `ENABLE_SIGSTP` isn't set. There's no need to fail so loudly - we can default to doing nothing. In fact, we can also remove the

```rs
#[cfg(not(windows))]
ENABLE_SIGSTP.set(false) // note: this also needs an .unwrap() or .ok()
```

in `application.rs` (comments below this one).

Overall, this can just be:

```suggestion
    if cfg!(not(windows)) && *crate::application::ENABLE_SIGTSTP.get() == Some(true) {
        signal_hook::low_level::raise(signal_hook::consts::signal::SIGTSTP).unwrap();
    }
```